### ISSUE TITLE
Failed to install pmdarima in Python 3.12

### ISSUE THREAD

**Issue by ferdcoz** [alkaline-ml/pmdarima#563]
### Describe the bug

I'm not able to install the package using Python v3.12

------------------------------------------------------------------------------

Building wheels for collected packages: pmdarima
  Building wheel for pmdarima (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for pmdarima (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [38 lines of output]
      <string>:15: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
      Partial import of pmdarima during the build process.

      Traceback (most recent call last):
        File "<string>", line 190, in check_package_status
        File "C:\Users\ferna\AppData\Local\Programs\Python\Python312\Lib\importlib\__init__.py", line 90, in import_module
          return _bootstrap._gcd_import(name[level:], package, level)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<frozen importlib._bootstrap>", line 1381, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1354, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1318, in _find_and_load_unlocked
      ModuleNotFoundError: No module named 'numpy'
      Requirements: ['joblib>=0.11\nCython>=0.29,!=0.29.18,!=0.29.31\nnumpy>=1.21.2\npandas>=0.19\nscikit-learn>=0.22\nscipy>=1.3.2\nstatsmodels>=0.13.2\nurllib3\nsetuptools>=38.6.0,!=50.0.0\n']
      Adding extra setuptools args
      Traceback (most recent call last):
        File "C:\Users\ferna\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Users\ferna\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\ferna\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 251, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\ferna\AppData\Local\Temp\pip-build-env-zfu7me61\overlay\Lib\site-packages\setuptools\build_meta.py", line 434, in build_wheel
          return self._build_with_temp_dir(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\ferna\AppData\Local\Temp\pip-build-env-zfu7me61\overlay\Lib\site-packages\setuptools\build_meta.py", line 419, in _build_with_temp_dir
          self.run_setup()
        File "C:\Users\ferna\AppData\Local\Temp\pip-build-env-zfu7me61\overlay\Lib\site-packages\setuptools\build_meta.py", line 507, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "C:\Users\ferna\AppData\Local\Temp\pip-build-env-zfu7me61\overlay\Lib\site-packages\setuptools\build_meta.py", line 341, in run_setup
          exec(code, locals())
        File "<string>", line 340, in <module>
        File "<string>", line 327, in do_setup
        File "<string>", line 210, in check_package_status
      ImportError: numpy is not installed.
      pmdarima requires numpy >= 1.16.

      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pmdarima
Failed to build pmdarima
ERROR: Could not build wheels for pmdarima, which is required to install pyproject.toml-based projects

### To Reproduce

1. - Set Python v3.12
2. run pip install pmdarima

### Versions

```shell
Not able to install it
```

### Expected Behavior

Get package installed

### Actual Behavior

Failling installation process.

### Additional Context

It's working perfectly fine for Python 3.9 (Anaconda) and 3.11

**Comment by aaronreidsmith** [comment]
Duplicate of #560 and #562. Fixed by #561. Plan is to publish a release tomorrow

**Comment by aaronreidsmith** [comment]
Python 3.12 support has been released: 

```bash
$ python -c 'import pmdarima; pmdarima.show_versions()'

System:
    python: 3.12.0 (main, Oct 12 2023, 00:42:11) [GCC 12.2.0]
executable: /usr/local/bin/python
   machine: Linux-6.4.16-linuxkit-aarch64-with-glibc2.36

Python dependencies:
 setuptools: 68.2.2
        pip: 23.2.1
    sklearn: 1.3.2
statsmodels: 0.14.0
      numpy: 1.26.1
      scipy: 1.11.3
     Cython: 3.0.4
     pandas: 2.1.1
     joblib: 1.3.2
   pmdarima: 2.0.4
```
